### ISSUE TITLE
Brunsli for JPG?

### ISSUE THREAD

**Issue by M-Gonzalo** [schnaader/precomp-cpp#97]
It's MIT licensed. And quite a bit faster than packJPG (a lot, actually, but a bit less strong). It is also under development, so it stands to reason we should recieve at least some improvements in the future... 

What do you think, Christian?

**Comment by schnaader** [comment]
Thanks for this advice, tested it and it is much faster indeed (size different is about 1% in my tests, but it was about 5 times faster both compressing and decompressing). I'll have a look at the code and try to integrate it. I'm not sure if it will replace packJPG, however. Most likely using packJPG will stay an option (for slower, but better JPG compression).

**Comment by andrew-epstein** [comment]
I personally am in favor of integrating it, while leaving in packJPG as well, and then having one of them the default and the other toggleable (doesn't matter to me which is default). @schnaader please feel free to let me know if you want any help integrating / testing.

**Comment by M-Gonzalo** [comment]
Agreed. I don't believe there is any need to undo all the work put into packJPG.

In my tests, there are a few images in which brunsli fails. Maybe packJPG could be handling those when it's not being used as the default method too. 

Thanks for the feedback BTW

**Comment by andrew-epstein** [comment]
Also, I found at least one, maybe more, issue(s) with packJPG by using clang's address sanitizer. I'll create GitHub issues when I have time.

**Comment by schnaader** [comment]
> In my tests, there are a few images in which brunsli fails.

@M-Gonzalo : Could you share those images? They would be useful for testing.

**Comment by M-Gonzalo** [comment]
> > In my tests, there are a few images in which brunsli fails.
> 
> @M-Gonzalo : Could you share those images? They would be useful for testing.

Sure thing! As soon as I find some shareable ones.

**Comment by andrew-epstein** [comment]
@M-Gonzalo how does Brunsli fail? It fails to make the file smaller, or it crashes?

**Comment by M-Gonzalo** [comment]
> @M-Gonzalo how does Brunsli fail? It fails to make the file smaller, or it crashes?

You know, I've been revisiting my quick tests to give you and Christian a proper answer, and I found that maybe it was a bug in the script what I saw. There was a couple of files without their .bru counterpart  so I just assumed it was because of a crash. Now I'm thinking there was no crash at all. Sorry, it was a quick and crude test while I was away at work. I'll be doing a lengthier analysis in the following weeks.
